src/panels/config/integrations/integration-panels/thread/dialog-thread-dataset.ts

```diff
@@ -0,0 +1,87 @@
+import { LitElement, html, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import { HassDialog } from "../../../../../dialogs/make-dialog-manager";
+import { HomeAssistant } from "../../../../../types";
+import { DialogThreadDatasetParams } from "./show-dialog-thread-dataset";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+
+@customElement("ha-dialog-thread-dataset")
+class DialogThreadDataset extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _params?: DialogThreadDatasetParams;
+
+  public async showDialog(
+    params: DialogThreadDatasetParams
+  ): Promise<Promise<void>> {
+    this._params = params;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._params) {
+      return nothing;
+    }
+    const network = this._params.network;
+    const dataset = network.dataset!;
+    const otbrInfo = this._params.otbrInfo;
+
+    const hasOTBR =
+      otbrInfo &&
+      dataset.extended_pan_id &&
+      otbrInfo.active_dataset_tlvs?.includes(dataset.extended_pan_id);
+
+    const canImportKeychain =
+      hasOTBR &&
+      !this.hass.auth.external?.config.canTransferThreadCredentialsToKeychain &&
+      network.routers?.length;
+
+    return html`<ha-dialog
+      open
+      .hideActions=${!canImportKeychain}
+      @closed=${this.closeDialog}
+      .heading=${createCloseHeading(this.hass, network.name)}
+    >
+      <div>
+        Network name: ${dataset.network_name}<br />
+        Channel: ${dataset.channel}<br />
+        Dataset id: ${dataset.dataset_id}<br />
+        Pan id: ${dataset.pan_id}<br />
+        Extended Pan id: ${dataset.extended_pan_id}<br />
+
+        ${hasOTBR
+          ? html`OTBR URL: ${otbrInfo.url}<br />
+              Active dataset TLVs: ${otbrInfo.active_dataset_tlvs}`
+          : nothing}
+      </div>
+      ${canImportKeychain
+        ? html`<ha-button slot="primary-action" @click=${this._sendCredentials}
+            >Send credentials to phone</ha-button
+          >`
+        : nothing}
+    </ha-dialog>`;
+  }
+
+  private _sendCredentials() {
+    this.hass.auth.external!.fireMessage({
+      type: "thread/store_in_platform_keychain",
+      payload: {
+        mac_extended_address:
+          this._params!.network.routers![0]!.extended_pan_id,
```

Comment:
is the extended_pan_id the mac extended address?

Reply:
Not sure

Reply:
Now that I am looking at this, I remember that @agners said in the past we would need to start saving the mac extended address (or was something else) for the OTBR, perhaps we are not saving yet?

Reply:
Don’t forget there might be a difference in the credentials with standard iOS and developer iOS builds. This was an issue @agners saw that he had to fix.

Reply:
What difference? If you mean the credentials that are already in iOS, they are the same for any build because it all comes from Apple Keychain

Reply:
> is the extended_pan_id the mac extended address?

In my case they were not the same for OTBR FYI. It does look like the router might have `extended_address` available in the router object.

https://github.com/home-assistant/frontend/blob/505d7b6ddba7f10eaf9cfd30fd677f04dc90258a/src/data/thread.ts#L7

Reply:
Yes, extended (MAC) address and extended PAN ID is not the same thing! So this line is wrong.

The extended address was always part of the OTBR object, so it should be available.

However, there was a change in how we reference the preferred border router: Unfortunately, Apple border routers do not use/have the Thread Border Agent ID, so this meant an Apple border router could not have been defined as a perferred border router. This changed with https://github.com/home-assistant/core/pull/109065 in the backend, and https://github.com/home-assistant/frontend/pull/19580 in the frontend. So all good there.